src/game/game.cpp

```diff
@@ -528,12 +528,32 @@ namespace game
 
     bool inzoom()
     {
-        if(lastzoom && (zooming || lastmillis-lastzoom <= W(focus->weapselect, cookzoom)))
-            return true;
-        return false;
+        return lastzoom && (zooming || lastmillis - lastzoom <= W(focus->weapselect, cookzoom));
     }
     ICOMMAND(0, iszooming, "", (), intret(inzoom() ? 1 : 0));
 
+    // Get the progress of the zoom in the interval [0, 1], where 0 is no zoom and 1 is full zoom.
+    float zoomratio(gameent *d)
+    {
+        int frame = lastmillis - lastzoom;
+
+        int cookzoom = W(d->weapselect, cookzoom);
+
+        float ratio = 1.f;
+        if (cookzoom > 0 && frame <= cookzoom) {
```

Comment:
Let's document that `cookzoom > 0` means `cooking enabled`. Then this default value also makes a hell lot more sense (as zooming is instantaneous when cooking is not enabled for a weapon).